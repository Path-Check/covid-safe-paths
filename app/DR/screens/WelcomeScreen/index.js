import React from 'react';
import { View, AsyncStorage } from 'react-native';
import { Text } from 'native-base';
import Instructions from '../../components/Instructions';
import styles from '../../components/styles';

function InsideInstructions({ title, text }) {
  return (
    <View style={styles.instructionsContent}>
      <Text style={styles.instructionsTitle}>{title}</Text>
      <Text style={styles.instructionsText}>{text}</Text>
    </View>
  );
}

export default function WelcomeScreen({ navigation }) {
  navigation.setOptions({
    headerShown: false,
  });

  storeData = async () => {
    try {
      await AsyncStorage.setItem('welcomeShown', 'yes');
    } catch (error) {
      console.log(error);
    }
  };

  React.useEffect(() => {
    const subscribe = navigation.addListener('focus', () => {
      storeData();
    });
    return subscribe;
  }, [navigation]);

  return (
    <Instructions
      navigation={navigation}
      data={[
        {
          image: require('../../assets/images/recommendations.jpg'),
          content: (
            <InsideInstructions
              title="Evalua tus síntomas"
              text="Si tienes algún síntoma parecido al COVID-19 lo evaluamos y te damos el seguimiento apropiado"
            />
          ),
        },
        {
          image: require('../../assets/images/news.jpg'),
          content: (
            <InsideInstructions
              title="Tranquilo e informado"
              text="Te mantendremos informado de lo que necesitas saber del COVID-19 en R.D. en un solo lugar"
            />
          ),
        },
        {
          image: require('../../assets/images/bulletins.jpg'),
          content: (
            <InsideInstructions
              title="Recibe Alertas"
              text="Recibe las alertas del Ministerio de Salud para mantener a tu familia asegurada"
            />
          ),
        },
      ]}
    />
  );
}
