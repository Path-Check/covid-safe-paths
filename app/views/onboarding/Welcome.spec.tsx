import React from 'react';
import {
  render,
  cleanup,
  wait,
  fireEvent,
} from '@testing-library/react-native';
import '@testing-library/jest-native/extend-expect';

import WelcomeScreen from './Welcome';

afterEach(cleanup);

describe('WelcomeScreen', () => {
  it('renders correctly', () => {
    const { asJSON } = render(<WelcomeScreen />);

    expect(asJSON()).toMatchSnapshot();
  });

  describe('when the user checks the eula checkbox', () => {
    it('allows the user to proceed', async () => {
      const { getByTestId } = render(<WelcomeScreen />);

      const checkbox = getByTestId('welcome-eula-checkbox');
      const getStartedButton = getByTestId('welcome-button');

      expect(getStartedButton).toBeDisabled();

      fireEvent.press(checkbox);

      await wait(() => {
        expect(getStartedButton).not.toBeDisabled();
      });
    });
  });
});
