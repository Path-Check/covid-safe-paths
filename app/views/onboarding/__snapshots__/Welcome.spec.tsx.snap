// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/images/launchScreenBackground.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
          "height": "100%",
          "resizeMode": "cover",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/images/launchScreenBackgroundOverlay.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "padding": 24,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.1)",
                "borderRadius": 100,
                "opacity": 1,
                "paddingHorizontal": 28,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2e2e2e",
                    "fontSize": 17,
                    "lineHeight": 28,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontSize": 15,
                    "lineHeight": 24,
                    "textTransform": "uppercase",
                  },
                ]
              }
            >
              English
            </Text>
          </View>
          <Modal
            animationType="slide"
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          >
            <View
              style={
                Object {
                  "flex": 2,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                style={
                  Object {
                    "backgroundColor": "#000000",
                    "flex": 1,
                    "opacity": 0.2,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
              >
                <View>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#2e2e2e",
                          "fontSize": 17,
                          "lineHeight": 28,
                        },
                        Object {
                          "writingDirection": "ltr",
                        },
                        Object {
                          "color": undefined,
                          "fontWeight": "bold",
                          "marginRight": 22,
                          "marginTop": 16,
                          "textAlign": "right",
                        },
                      ]
                    }
                  >
                    Done
                  </Text>
                </View>
                <View>
                  <View>
                    <RCTPicker
                      items={
                        Array [
                          Object {
                            "label": "English",
                            "textColor": undefined,
                            "value": "en",
                          },
                        ]
                      }
                      selectedIndex={0}
                      style={
                        Array [
                          Object {
                            "height": 216,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </Modal>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flex": 5,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2e2e2e",
                  "fontSize": 17,
                  "lineHeight": 28,
                },
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "IBMPlexSans-Bold",
                  "fontSize": 28,
                  "fontWeight": "500",
                  "lineHeight": 40,
                },
              ]
            }
          >
            Welcome to PathCheck
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 2,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingBottom": 20,
              }
            }
          >
            <View
              accessibilityLabel="I accept the Terms of Use."
              accessibilityRole="checkbox"
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
              testID="welcome-eula-checkbox"
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../app/assets/images/boxUncheckedIcon.png",
                  }
                }
                style={
                  Object {
                    "height": 25,
                    "marginRight": 20,
                    "width": 25,
                  }
                }
              />
            </View>
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2e2e2e",
                      "fontSize": 17,
                      "lineHeight": 28,
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontSize": 15,
                      "lineHeight": 20,
                    },
                  ]
                }
              >
                I accept the Terms of Use.
              </Text>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2e2e2e",
                        "fontSize": 17,
                        "lineHeight": 28,
                      },
                      Object {
                        "writingDirection": "ltr",
                      },
                      Object {
                        "color": "#ffcc00",
                        "fontSize": 17,
                        "lineHeight": 28,
                        "textDecorationLine": "underline",
                      },
                    ]
                  }
                  testID="welcome-termsofuse-link"
                >
                  Terms of Use
                </Text>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255, 255, 255, 0.65)",
                "borderColor": "#ffffff",
                "borderRadius": 8,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 25,
                "paddingTop": 24,
              }
            }
            testID="welcome-button"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2e2e2e",
                    "fontSize": 17,
                    "lineHeight": 28,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "#1f2c9b",
                    "fontSize": 19,
                    "fontWeight": "500",
                  },
                ]
              }
            >
              Get Started
            </Text>
          </View>
        </View>
        <Modal
          animationType="slide"
          hardwareAccelerated={false}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "color": "#2e2e2e",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "space-between",
              }
            }
          >
            <RCTSafeAreaView
              emulateUnlessSupported={true}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 7,
                    "paddingHorizontal": 16,
                  }
                }
              >
                <View
                  accessibilityLabel="Close"
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  style={
                    Object {
                      "alignContent": "center",
                      "alignItems": "center",
                      "alignSelf": "flex-end",
                      "opacity": 1,
                      "padding": 12,
                    }
                  }
                >
                  <SvgXml
                    height={20}
                    style={Object {}}
                    width={20}
                    xml="
<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"#1f2c9b\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <path d=\\"M3.25731 22.5795L3.25737 22.5796L3.26062 22.5762L11.9494 13.3715L20.6888 22.6609L20.6887 22.661L20.691 22.6632C20.9628 22.9377 21.2962 23.125 21.789 23.125C22.4855 23.125 23.125 22.6697 23.125 22.0232C23.125 21.6762 22.9451 21.4335 22.7374 21.215L13.7371 11.7456L22.3837 2.74296L22.3842 2.74249C22.6022 2.51308 22.7213 2.27133 22.7213 1.97683C22.7213 1.38538 22.1473 0.875 21.4862 0.875C20.9837 0.875 20.6509 1.11422 20.339 1.42046L20.339 1.42041L20.3359 1.42364L12.0506 10.1618L3.71471 1.33886L3.71474 1.33884L3.71268 1.33676C3.4409 1.06226 3.10745 0.875 2.61468 0.875C1.91817 0.875 1.27867 1.33027 1.27867 1.97683C1.27867 2.32383 1.45862 2.56647 1.66628 2.78496L1.66648 2.78517L10.2629 11.7876L1.21239 21.2573L1.21215 21.2575C0.994108 21.4869 0.875 21.7287 0.875 22.0232C0.875 22.6146 1.44905 23.125 2.11009 23.125C2.61264 23.125 2.94544 22.8858 3.25731 22.5795Z\\" stroke-width=\\"0.25\\"/>
</svg>
"
                  />
                </View>
              </View>
            </RCTSafeAreaView>
          </View>
        </Modal>
      </View>
    </View>
  </View>
</View>
`;
